lib/widget/camera.dart

```diff
@@ -1,34 +1,75 @@
 
 import 'package:camera/camera.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
 import 'package:flutter/material.dart';
 import 'package:fluttertoast/fluttertoast.dart';
 import 'package:image_picker/image_picker.dart';
 // import 'package:flutter/foundation.dart' show kIsWeb;
 
-class CameraScreen extends StatefulWidget {
-  const CameraScreen({
+class CameraScreen extends StatefulWidget with Invokable, HasController<MyCameraController, CameraScreenState>{
+  static const type = 'Camera';
+  CameraScreen({
     Key? key,
-    this.mode = 'photo',
-    this.initialCamera = 'back',
-    this.useGallery = true,
-    this.maxCount = 1,
-    this.preview = false,
+    // this.mode = 'photo',
+    // this.initialCamera = 'back',
+    // this.useGallery = true,
+    // this.maxCount = 1,
+    // this.preview = false,
   }) : super(key: key);
 
-  final String mode;
-  final String initialCamera;
-  final bool useGallery;
-  final int maxCount;
-  final bool preview;
+  // final String mode;
+  // final String initialCamera;
+  // final bool useGallery;
+  // final int maxCount;
+  // final bool preview;
+
+  final MyCameraController _controller = MyCameraController();
+
+  @override
+  State<StatefulWidget> createState() => CameraScreenState();
+
+  @override
+  MyCameraController get controller => _controller;
+
+  @override
+  Map<String, Function> getters() {
+    return {
+
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
 
   @override
-  State<CameraScreen> createState() => _CameraScreenState();
+  Map<String, Function> setters() {
+    return {
+      'mode':(value) => _controller.mode = Utils.optionalString(value) ?? _controller.mode,
```

Comment:
don't do this. Make it explicit like optionalString(value) ?? CameraMode.BOTH

Reply:
yes i change string to enum and define according to your requirements